app/main.py

```diff
@@ -1,6 +1,32 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Any:
+        is_cachable = True
+        if kwargs:
+            is_cachable = False
+        else:
+            for arg in args:
+                if not isinstance(
+                        arg, (bool, int, float, str, tuple, frozenset, bytes)
+                ):
+                    is_cachable = False
```

Comment:
In this task it isn't mentioned that you can get kwargs, but if you decided to handle this case, then one note that it would be better to use "if isinstance" and set check mutable types. Easier to read and you see what types are mutable